moduleframework/helpers/openshift_helper.py

```diff
@@ -0,0 +1,300 @@
+                        pod_initiated = True
+                        break
+            if pod_initiated:
+                break
+        return pod_initiated
+
+    def setUp(self):
+        """
+        It is called by child class and it is same methof as Avocado/Unittest has. It prepares environment
+        for docker testing
+        * start docker if not
+        * pull docker image
+        * setup environment from config
+        * run and store identification
+
+        :return: None
+        """
+        self.icontainer = self.get_url()
+
+    def _openshift_login(self, oc_ip="127.0.0.1", oc_user='developer', oc_passwd='developer', env=False):
+        """
+        It logins to an OpenShift environment on specific IP and under user and his password.
+        :param oc_ip: an IP where is an OpenShift environment running
+        :param oc_user: an username under which we can login to OpenShift environment
+        :param oc_passwd: a password for specific username
+        :param env: is used for specification OpenShift IP, user and password, otherwise defaults are used
+        :return:
+        """
+        if env:
+            oc_ip = common.get_openshift_ip()
+            oc_user = common.get_openshift_user()
+            oc_passwd = common.get_openshift_passwd()
+        oc_output = self.runHost("oc login %s:8443 --username=%s --password=%s" % (oc_ip,
+                                                                                   oc_user,
+                                                                                   oc_passwd),
+                                 verbose=common.is_not_silent())
+        return oc_output.exit_status
+
+    def tearDown(self):
+        """
+        Cleanup environment and call also cleanup from config
+
+        :return: None
+        """
+        super(OpenShiftHelper, self).tearDown()
+        try:
+            self._app_remove()
+        except Exception as e:
+            common.print_info(e, "OpenShift application already removed")
+            pass
+
+    def _get_ip_instance(self):
+        """
+        It gets and IP address of an application from of OpenShift POD.
+        :return: True: getting IP address was successful
+                 False: getting IP address was not successful
+        """
+        oc_get_service = self.runHost("oc get service -o json")
+        service = self._convert_string_to_json(oc_get_service.stdout)
+        try:
+            for svc in service:
+                if svc.get('metadata').get('labels').get('app') == self.app_name:
+                    self.ipaddr = svc.get('spec').get("clusterIP")
+                    common.trans_dict['GUESTIPADDR'] = self.ipaddr
+            return True
+        except KeyError as e:
+            common.print_info(e.message)
+            return False
+        except IndexError as e:
+            common.print_info(e.message)
+            return False
+
+    def getIPaddr(self):
```

Comment:
pythonic way of doing setters and getters is like this: https://docs.python.org/3/library/functions.html#property

(not blocking this review, just a friendly advice)

Reply:
https://github.com/fedora-modularity/meta-test-family/issues/167